tests/framework/e2e/etcd_process.go

```diff
@@ -375,7 +375,11 @@ func (f *BinaryFailpoints) SetupHTTP(ctx context.Context, failpoint, payload str
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusNoContent {
-		return fmt.Errorf("bad status code: %d", resp.StatusCode)
+		errMsg, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("bad status code: %d (%v)", resp.StatusCode, err)
+		}
+		return fmt.Errorf("bad status code: %d (%v)", resp.StatusCode, errMsg)
```

Comment:
I think we should use `string(errMsg)` here becasue `errMsg` is `[]byte`. It's not human-friendly.

Reply:
Or just use `%s`, see https://github.com/etcd-io/etcd/pull/17878

Reply:
Fixed accordingly!

Also fixed the err's formatting to using %w